src/dbnode/storage/series/buffer.go

```diff
@@ -281,6 +311,26 @@ func (b *dbBuffer) IsEmpty() bool {
 	return len(b.bucketsMap) == 0
 }
 
+func (b *dbBuffer) ColdFlushBlockStarts(latestWarmFlushTime time.Time) OptimizedTimes {
+	var times OptimizedTimes
+
+	for t, bucketVersions := range b.bucketsMap {
+		for _, bucket := range bucketVersions.buckets {
+			if bucket.version == writableBucketVersion &&
+				bucket.writeType == ColdWrite &&
+				// Cold flushes can only happen on blockStarts that have been
+				// warm flushed, because warm flush logic does not currently
+				// perform any merging logic.
+				!bucket.start.After(latestWarmFlushTime) {
```

Comment:
Are we gonna remove this check? Seems weird to have it here

Reply:
Yep, will remove this check and use shard flush state to determine whether something has been warm flushed instead.